app/models/comment.rb

```diff
@@ -217,7 +217,8 @@ def shorten_urls!
       urls.each do |url|
         anchor_content.sub!(/#{Regexp.escape(url)}/, strip_url(url))
       end
-      anchor.inner_html = anchor.inner_html.sub!(/#{Regexp.escape(anchor.content)}/, anchor_content)
+      shortened_content = anchor.inner_html.sub!(/#{Regexp.escape(anchor.content)}/, anchor_content)
+      anchor.inner_html = shortened_content if shortened_content.present?
```

Comment:
If we're assigning the result it seems a bit inappropriate to use an in-place mutating method in the first place. `sub` doesn't return `nil` if no modification happens so won't need an extra `if` check.

```ruby
"test".sub!('a', 'x')
#=> nil
"test".sub('a', 'x')
#=> "test"
```

```suggestion
      anchor.inner_html = anchor.inner_html.sub(/#{Regexp.escape(anchor.content)}/, anchor_content)
```

Both the original spec (`"shortens long urls without removing formatting"`) and your new one are green with this.